src/bentoml/_internal/cloud/deployment.py

```diff
@@ -1,73 +1,222 @@
+        return bentoml_cattr.structure(kwargs, DeploymentTargetConfig)
+
+
+class Deployment:
+    def __init__(self) -> None:
+        self.resource = Resource()
+
+    def _create_deployment(
+        self,
+        context: str | None = None,
+        cluster_name: str | None = None,
+        *,
+        create_deployment_schema: CreateDeploymentSchema,
+    ) -> DeploymentSchema:
+
+        yatai_rest_client = get_rest_api_client(context)
+        if cluster_name is None:
+            cluster_name = default_context_name
+        for target in create_deployment_schema.targets:
+            res = yatai_rest_client.get_bento(target.bento_repository, target.bento)
+            if res is None:
+                raise BentoMLException(
+                    f"Create deployment: {target.bento_repository}:{target.bento} does not exist"
+                )
+        res = yatai_rest_client.get_deployment(
+            cluster_name,
+            create_deployment_schema.kube_namespace,
+            create_deployment_schema.name,
+        )
+        if res is not None:
+            raise BentoMLException("Create deployment: Deployment already exists")
+        res = yatai_rest_client.create_deployment(
+            cluster_name, schema_to_json(create_deployment_schema)
+        )
+        if res is None:
+            raise BentoMLException("Create deployment request failed")
+        print(f"{create_deployment_schema.name} is created.")
+        return res
+
+    def create_deployment(
+        self,
+        deployment_name: str,
+        bento_repository: Tag | str,
+        description: str = None,
+        cluster_name: str = default_context_name,
+        kube_namespace: str = default_kube_namespace,
+        cpu: str | None = None,
+        gpu: str | None = None,
+        memory: str | None = None,
+        hpa_conf: DeploymentTargetHPAConf | None = None,
+        runners_config: dict[str, DeploymentTargetRunnerConfig] | None = None,
+        api_server_config: DeploymentTargetConfig | None = None,
+        mode: DeploymentMode = DeploymentMode.Function,
+        type: DeploymentTargetType = DeploymentTargetType.STABLE,
+        context: str | None = None,
+        do_not_deploy: bool = False,
+        labels: t.List[dict[str, str]] | None = None,
+        canary_rules: t.List[DeploymentTargetCanaryRule] | None = None,
+    ) -> DeploymentSchema:
+        if isinstance(bento_repository, str):
+            bento_repository = bentoml_cattr.structure(bento_repository, Tag)
+        print(bento_repository)
+        dct = {
+            "name": deployment_name,
+            "kube_namespace": kube_namespace,
+            "mode": mode,
+            "labels": labels,
+            "description": description,
+            "do_not_deploy": do_not_deploy,
+        }
+        if api_server_config is None:
+            dct["targets"] = [
+                {
+                    "type": type,
+                    "bento_repository": bento_repository.name,
```

Comment:
What happens here if cpu gpu or mem is not None?

Reply:
Umm should parameter override the config info? Normally user would either have the parameters filled or the api_server_config filled. In case of both are filled, should we prioritize the parameters?

Reply:
I think either works, depending on the order. I think we can do a hierarchy parsing,

if both params and the dict pass, then the dict will be the final value